src/components/TodoList/TodoList.tsx

```diff
@@ -1,100 +1,97 @@
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">quis ut nam facilis et officia qui</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye-slash" />
-            </span>
-          </button>
-        </td>
-      </tr>
+const filteredTodosList = (value: string, todos: Todo[]) => {
+  return todos.filter(todo => todo.title
+    .toLocaleLowerCase().includes(value.trim().toLocaleLowerCase()));
+};
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">1</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">delectus aut autem</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
+export const TodoList: React.FC<ListOfTodos>
+= ({
+  todoList, modalActive, query, option, activeTaskId,
+}) => {
+  const isDoneStyle = (isDone: boolean) => {
+    return classNames({
+      'has-text-danger': !isDone,
+      'has-text-success': isDone,
+    });
+  };
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">6</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">
-            qui ullam ratione quibusdam voluptatem quia omnis
-          </p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
+  let filteredTodos: Todo[] = [];
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">8</td>
-        <td className="is-vcentered">
-          <span className="icon" data-cy="iconCompleted">
-            <i className="fas fa-check" />
-          </span>
-        </td>
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-success">quo adipisci enim quam ut ab</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
+  if (option === null) {
+    filteredTodos = filteredTodosList(query, todoList);
+  } else if (option) {
+    filteredTodos
+     = filteredTodosList(query, todoList)
+        .filter(todo => todo.completed === true);
+  } else {
+    filteredTodos
+    = filteredTodosList(query, todoList)
+        .filter(todo => todo.completed === false);
+  }
+
```

Comment:
This would fit perfectly into `useMemo` read about it [here](https://react.dev/reference/react/useMemo)